### PR TITLE
Reland Enable MSAA behind a flag for iOS

### PR DESCRIPTION
**Pull request by dnfield**
Relands #33461 

The patch was previously reverted because of a mistake in the embedder code: I had transposed the position of the `render_to_surface` argument and the `msaa_samples` argument, and C++ did an implicit int to bool conversion which broke everything on desktop.

I've manually verified that desktop works, and I've switched to using an enum _everywhere_ for the MSAA sample count argument so that the type error won't happen again.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
RSLGTM

File: shell/platform/embedder/embedder_surface_metal.mm [hunk 1]
```diff
@@ -44,7 +44,8 @@
   }
 
   const bool render_to_surface = !external_view_embedder_;
-  auto surface = std::make_unique<GPUSurfaceMetalSkia>(this, main_context_, render_to_surface);
+  auto surface = std::make_unique<GPUSurfaceMetalSkia>(this, main_context_, MsaaSampleCount::kNone,
+                                                       render_to_surface);
```
Comment (dnfield) [inline shell/platform/embedder/embedder_surface_metal.mm:48]: Specifically, in the previous version of this patch I wrote

```cpp
auto surface = std::make_unique<GPUSurfaceMetalSkia>(this, main_context_, render_to_surface, 1);
```

Which meant `render_to_surface` was _always_ true, which is the wrong thing to do (and caused the blank screen rendering).

Comment (dnfield) [reply shell/platform/embedder/embedder_surface_metal.mm:48]: Using the enum everywhere means that wouldn't even compile now.

File: shell/platform/darwin/ios/framework/Source/FlutterEnginePlatformViewTest.mm [hunk 2]
```diff
@@ -87,6 +87,29 @@ - (void)tearDown {
   return weak_factory->GetWeakPtr();
 }
 
+- (void)testMsaaSampleCount {
```
Comment (jonahwilliams) [inline shell/platform/darwin/ios/framework/Source/FlutterEnginePlatformViewTest.mm:90]: Test needs to be updated to use the enum

**Pull request merged**